lib/common.sh

```diff
@@ -789,6 +789,50 @@ common::print_n_char () {
 }
 
 ###############################################################################
+# Generate a (github) markdown TOC between BEGIN/END tags
+# @param file The file to update in place
+#
+common::mdtoc () {
+  local file=$1
+  local indent
+  local anchor
+  local heading
+  local begin_block="<!-- BEGIN MUNGE: GENERATED_TOC -->"
+  local end_block="<!-- END MUNGE: GENERATED_TOC -->"
+  local tmpfile=/tmp/$PROG-cm.$$
+
+  declare -A count
+
+  while read level heading; do
+    indent="$(echo $level |sed -e "s,^#,,g" -e 's,#,  ,g')"
+    # make a valid anchor
+    anchor=${heading,,}
+    anchor=${anchor// /-}
+    anchor=${anchor//[\.\?\*\,\/()]/}
+    # Keep track of dups and identify
+    if [[ -n ${count[$anchor]} ]]; then
+      ((count[$anchor]++)) ||true
+      ((${count[$anchor]}>0)) && anchor+="-${count[$anchor]}"
```

Comment:
I think this conditional might not be necessary - `$count[$anchor]` should always be greater than zero, since it was incremented on the previous line, right?